httpx/_models.py

```diff
@@ -481,9 +481,6 @@ def update(self, headers: HeaderTypes = None) -> None:  # type: ignore
         for header in headers:
             self[header] = headers[header]
 
-    def copy(self) -> "Headers":
-        return Headers(self.items(), encoding=self.encoding)
-
```

Comment:
Did you mean to remove this? Evidently we're not using it right now, but we *probably* still want it as part of the public API. (Or at least, if we *did* want to remove it, then it's a *small* breaking change, so we'd need to do so in a major version bump, so we can call it out in the CHANGELOG)

Reply:
I'd prefer to keep it altogether, I feel a lot of our users to header manipulation and rely on this method.

Reply:
Hey @tomchristie, yes, I removed it on purpose since it isn't documented and I thought it was just used internally: https://www.python-httpx.org/api/#url

Reply:
Also I noticed the same thing can be done already with the class constructor, but happy to roll this change back if you prefer 🙂 
https://github.com/encode/httpx/blob/21d7e16559d9360ae3a5c5cfd23bab8bb85ee4a8/httpx/_models.py#L391

Reply:
I've added the method back and documented & tested it.